sqflite integration for Flutter (#6500)
getsentry/sentry-docs: src/platforms/flutter/common/configuration/integrations/sqflite-instrumentation.mdx

---
title: sqflite Database Instrumentation
caseStyle: camelCase
supportLevel: production
sdk: sentry.dart.sqflite
description: "Learn more about the Sentry sqflite Database Instrumentation for the Flutter SDK."
categories:
  - mobile
---

> <Include name="beta-note.mdx" />

Reviewer: `m`: do we actually beta this? if so, do we usually use `supportLevel: production` for beta integrations?

Author: `supportLevel` is only `production` or `community, and not really related to the support level similar to semver.

Author: I'd keep it beta at least for a few days/weeks since we use some internal APIs, we need to be sure that we are not breaking any behavior for any platform.

_(New in version 7.2.0)_

> The [sentry_sqflite](https://pub.dev/packages/sentry_sqflite) package provides `sqflite` support for Database Instrumentation. The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/sqflite).

Reviewer: ```suggestion
The [sentry_sqflite](https://pub.dev/packages/sentry_sqflite) package provides `sqflite` support for database instrumentation. The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/sqflite).
```

import 'package:sentry_sqflite/sentry_sqflite.dart';
import 'package:sqflite/sqflite.dart';

databaseFactory = SentrySqfliteDatabaseFactory();
final database = await openDatabase('path/to/db');
```

If you're using the `FFI` factories, you can use the `SentrySqfliteDatabaseFactory` with its factory:

```dart
import 'package:sentry_sqflite/sentry_sqflite.dart';
import 'package:sqflite/sqflite.dart';

databaseFactory = SentrySqfliteDatabaseFactory(databaseFactory: databaseFactoryFfi);
final database = await openDatabase('path/to/db');
```

Using the `openDatabaseWithSentry` wrapper, only the database opened with this method will be instrumented:

```dart
import 'package:sentry_sqflite/sentry_sqflite.dart';
import 'package:sqflite/sqflite.dart';

final database = await openDatabaseWithSentry('path/to/db');
// or final database = await openReadOnlyDatabaseWithSentry('path/to/db');
```

Using the `SentryDatabase` wrapper, which can be used to instrument any `Database` instance:

```dart
import 'package:sentry_sqflite/sentry_sqflite.dart';
import 'package:sqflite/sqflite.dart';

final database = await openDatabase('path/to/db');
final sentryDatabase = SentryDatabase(database);
```

> After doing this, every `CRUD` operation including transactions and batches will be automatically instrumented and sent to Sentry.

Reviewer: ```suggestion
After doing this, every `CRUD` operation including database transactions and batches will be automatically instrumented and sent to Sentry.
```

> The span's `operation`'s are either `db`, `db.sql.execute`, `db.sql.query` or `db.sql.transaction` and it's `description` is the SQL statement using placeholders instead of its values due to the possibility of containing PII.

Reviewer: ```suggestion
The spans' `operation`s are either `db`, `db.sql.execute`, `db.sql.query` or `db.sql.transaction`. Its `description` is the SQL statement using placeholders instead of its values due to the possibility of containing PII.
```
